### PR TITLE
Create Endpoints for Login with Passport

### PR DESCRIPTION
**Pull request by jciolfi**
Will add comments & clean up code later

### CODE REVIEW

File: repositories/user.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import UserSchema from '../dbmodels/user';
+import HTTPError from '../exceptions/HTTPError';
+import { IUser } from '../types/dbtypes/user';
+import { Schema } from 'mongoose';
+
+export const get_or_create_user = async (
```
Comment (nicoledanuwidjaja) [inline repositories/user.ts:6]: Smh snake case

File: passport.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          })
+          .catch((err) => {
+            done(err, undefined);
+            return;
+          });
+      },
+    ),
+  );
+
+  // Auth Routes
```
Comment (nicoledanuwidjaja) [inline passport.ts:100]: It would be nice to have this in `/routes` under an authentication route file, such as `auth.ts`.

**Review by nicoledanuwidjaja — APPROVED** [review]
Only took a look at the passport related stuff, and the logic looks sound to me! Just a comment for project structuring

**Review by maxpinheiro — CHANGES_REQUESTED** [review]
looks good, one small thing - I tried to start the server without filling out the .env and received a somewhat confusing error message:
`[ERROR] 20:27:28 TypeError: OAuth2Strategy requires a clientID option`
if you can make this error message more clear, you're good to go

**Review by maxpinheiro — APPROVED** [review]
lgtm 👍

**Pull request merged**